frame/benchmarking/src/lib.rs

```diff
@@ -911,14 +970,21 @@ macro_rules! impl_benchmark {
 	};
 	(
 		{ $( $where_clause:tt )* }
-		INSTANCE $( $name:ident ),*
+		INSTANCE
+		( $( $name:ident ),* )
+		( $( $name_extra:ident ),* )
 	) => {
 		impl<T: Trait<I>, I: Instance> $crate::Benchmarking<$crate::BenchmarkResults>
 			for Module<T, I>
 			where T: frame_system::Trait, $( $where_clause )*
 		{
-			fn benchmarks() -> Vec<&'static [u8]> {
-				vec![ $( stringify!($name).as_ref() ),* ]
+			fn benchmarks(extra: bool) -> Vec<&'static [u8]> {
+				let mut all = vec![ $( stringify!($name).as_ref() ),* ];
+				if !extra {
+					let extra = vec![ $( stringify!($name_extra).as_ref() ),* ];
```

Comment:
```suggestion
					let extra = [ $( stringify!($name_extra).as_ref() ),* ];
```

Some as above. Would be nice if we could get rid of all that copy & paste code duplication.

Reply:
Yeah, I actually asked for @thiolliere 's help here.

There is basically 2 copies of all the code, one for Pallet with instance `<T, I>` and regular pallet `<T>`.

There is probably some way to do this with macros, but I have tried, and I could not abstract at the right layer to make it work.